raiden-ts/src/transport/epics/init.ts

```diff
@@ -390,10 +389,10 @@ export const matrixShutdownEpic = (
     mergeMap((matrix) =>
       action$.pipe(
         finalize(() => {
+          matrix.stopClient();
           matrix.setPresence({ presence: 'offline', status_msg: '' }).catch(() => {
```

Comment:
Does it make sense to stop the client and then send a presence update?

Also, could this cause unresolved promises in e.g. our tests? Looks like `setPresence` returns a promise that we don't await for so it could be pending for quite a bit

Reply:
`stopClient` stops the `sync` loop which makes us appear online. stopping it after marking the offline status could in rare race conditions make us appear online for another 30s (next loop), therefore I stop it before the `setPresence`. The `setPresence` call is a simple http endpoint, used just to ensure we're marked earlier as offline (usually, we'd appear as online until server doesn't see sync requests for another 30s, which could make peers keep retrying something with us while we're not listening anymore). This shouldn't cause unresolved promises in our tests, specially because we mock matrix and this endpoint should resolve immediately, while tests only complain if some promise isn't resolved 1s after tests end. We can't await anyway here, since it's a `finalize` call, performed in the subscription's teardown/shutdown/sync stop, so it's a 'best effort' call.

Reply:
We don't mock matrix in integration tests

Reply:
Yes, I know, but there it's a local server which should reply on less than 1s. Anyway, this doesn't change the previous behavior, which did work fine, just move the call a little to prevent an hypothetical race on offline/sync loop which wouldn't cause any major issue than a small delay on clients on detecting our status.